### PR TITLE
Basic text input and tests

### PR DESCRIPTION
**Pull request by B1naryVision**

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -37,8 +37,20 @@ module.exports = function (config) {
     colors: true,
     logLevel: config.LOG_INFO,
     autoWatch: true,
-    browsers: ['Chrome'],
+    browsers: ['ChromeHeadless'],
```
Comment (B1naryVision) [inline karma.conf.js:40]: Hopefully you don't mind, can undo, but I'm running everything via CLI so cant actually run chrome.

File: package.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
     "karma-coverage": "~2.0.3",
     "karma-jasmine": "~4.0.0",
     "karma-jasmine-html-reporter": "^1.5.0",
+    "ng-mocks": "^12.3.1",
```
Comment (B1naryVision) [inline package.json:37]: Spent some time mocking my component, not hard but looks awful and re-usable components are sure to have duplicates.

This works well assuming we're good with Shallow copies, to keep tests strictly to the unit and ignoring child behaviour excluding what it tells the parent.

File: src/assets/utilities/grid.css [hunk 3]
```diff
@@ -0,0 +1,84 @@
+.grid {
```
Comment (B1naryVision) [inline src/assets/utilities/grid.css:1]: I saw you added Bootstrap in your latest branch, wishI had seen that... I will undo my usages of this one that is in the repo.

File: src/styles.css [hunk 4]
```diff
@@ -1 +1,8 @@
 /* You can add global styles to this file, and also import other style files */
+@import "assets/utilities/grid.css";
+@import "assets/utilities/padding.css";
+
+body {
+  background-color: #333333;
+  color: #fff;
```
Comment (B1naryVision) [inline src/styles.css:7]: creative liberties, complain if hate

File: karma.conf.js (hunk 1 above)
Comment (Gibbonator) [reply karma.conf.js:40]: I don't mind - I assume the literal visual rendering time if somewhat accounted for. Unless we get an issue that the tests with headless aren't picking up, I think that's dandy

File: src/styles.css (hunk 4 above)
Comment (Gibbonator) [reply src/styles.css:7]: I am not in a position to criticise :D

File: src/app/messages/messages.component.ts [hunk 5]
```diff
@@ -2,14 +2,20 @@ import { Component, OnInit } from '@angular/core';
...
   constructor() { }
 
   ngOnInit(): void {
+    // @todo load previous messages
   }
 
+  handleSend(textContent: string) {
+    this.messages = [...this.messages, { textContent, createdDate: new Date() }];
+
+    // @todo send to backend with User ID/Name
```
Comment (B1naryVision) [inline src/app/messages/messages.component.ts:19]: On second thought, I'm being a doofus here. Should send it to backend first as we need to know if it actually sent, but still display it. That way we can allow the user to click retry if it failed.

so need to add message status field.

**Review by Gibbonator — APPROVED** [review]
Found the approve button

**Pull request merged**